plugins/verification/src/main/java/org/qbicc/plugin/verification/ClassInitializingBasicBlockBuilder.java

```diff
@@ -0,0 +1,97 @@
+package org.qbicc.plugin.verification;
+
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.BasicBlockBuilder;
+import org.qbicc.graph.DelegatingBasicBlockBuilder;
+import org.qbicc.graph.Value;
+import org.qbicc.graph.ValueHandle;
+import org.qbicc.type.ClassObjectType;
+import org.qbicc.type.definition.DefinedTypeDefinition;
+import org.qbicc.type.definition.LoadedTypeDefinition;
+import org.qbicc.type.definition.element.ConstructorElement;
+import org.qbicc.type.definition.element.FieldElement;
+import org.qbicc.type.definition.element.InitializerElement;
+import org.qbicc.type.definition.element.MethodElement;
+
+/**
+ *
+ */
+public class ClassInitializingBasicBlockBuilder extends DelegatingBasicBlockBuilder {
+
+    private final CompilationContext ctxt;
+
+    public ClassInitializingBasicBlockBuilder(final CompilationContext ctxt, final BasicBlockBuilder delegate) {
+        super(delegate);
+        this.ctxt = ctxt;
+    }
+
+    @Override
+    public ValueHandle instanceFieldOf(ValueHandle instance, FieldElement field) {
+        initialize(field.getEnclosingType());
+        return super.instanceFieldOf(instance, field);
+    }
+
+    @Override
+    public ValueHandle staticField(FieldElement field) {
+        initialize(field.getEnclosingType());
```

Comment:
There are two cases for static fields and static methods - if the method is defined in an interface, we only initailize that interface.  If it's on a class, then we walk the superclasses and superinterfaces.

Is the intention that this over-registers code and we trim it later?  Should this avoid adding unnecessary initializers?

Reply:
No, that wasn't intended. I can tighten it up for static fields and methods of interfaces.

Reply:
Fixed.